docs/sync-rules/redis.md

```diff
@@ -0,0 +1,103 @@
+### Setting up the Redis server/cloud connector
+
+See the [Developer guide](../../docs/DEVELOPING.md) for setting up connectors.
+
+### Overview
+
+Advanced Sync Rules help manage data in Redis server/cloud. They take the following parameters:
+
+1. `database`: Database indice of Redis server/cloud. Type of value should be integer.
+2. `key_pattern`: Pattern of the key for searching in Redis server/cloud
+3. `_type`: Type of key(i.e. HASH, LIST, SET, STREAM, STRING, ZSET) in Redis server/cloud.
```

Comment:
```suggestion
3. `_type`: Type of key in Redis server/cloud. Supported values are HASH, LIST, SET, STREAM, STRING, ZSET
```

Make supported values explicit

I actually had doubts when I reviewed https://github.com/elastic/connectors/pull/2044, seeing a method param starting with `_`. I give it a pass because I see `_type` is also used in [scan_iter](https://redis-py.readthedocs.io/en/stable/commands.html#redis.commands.core.CoreCommands.scan_iter)

But we should not define `_type` as a rule field. Or we should also consider rename all the `_type` to `type` in https://github.com/elastic/connectors/pull/2044

Reply:
But `type` is a built-in in python. can we use something like `key_type` instead of using built-in then?

Reply:
You can suffix it, `type_`